pkgs/by-name/fl/flac/package.nix

```diff
@@ -1,62 +1,74 @@
 {
-  lib,
-  stdenv,
-  fetchurl,
   cmake,
-  pkg-config,
   doxygen,
+  fetchFromGitHub,
   graphviz,
+  lib,
   libogg,
+  nix-update-script,
+  pandoc,
+  pkg-config,
+  stdenv,
+  versionCheckHook,
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "flac";
-  version = "1.4.3";
+  version = "1.5.0";
 
-  src = fetchurl {
-    url = "http://downloads.xiph.org/releases/flac/${pname}-${version}.tar.xz";
-    # Official checksum is published at https://github.com/xiph/flac/releases/tag/${version}
-    hash = "sha256-bFjmnNIjSPRBuGEJK4JeWR0Lgi4QbebrDuTQXScgW3A=";
+  src = fetchFromGitHub {
+    owner = "xiph";
+    repo = "flac";
+    tag = finalAttrs.version;
+    hash = "sha256-B6XRai5UOAtY/7JXNbI3YuBgazi1Xd2ZOs6vvLq9LIs=";
   };
 
   nativeBuildInputs = [
     cmake
-    pkg-config
     doxygen
     graphviz
+    pandoc
+    pkg-config
   ];
 
-  buildInputs = [
-    libogg
-  ];
+  buildInputs = [ libogg ];
 
-  cmakeFlags = lib.optionals (!stdenv.hostPlatform.isStatic) [
-    "-DBUILD_SHARED_LIBS=ON"
-  ];
+  cmakeFlags = lib.optionals (!stdenv.hostPlatform.isStatic) [ "-DBUILD_SHARED_LIBS=ON" ];
 
   CFLAGS = [
     "-O3"
     "-funroll-loops"
   ];
   CXXFLAGS = [ "-O3" ];
 
-  # doCheck = true; # takes lots of time
+  doCheck = false; # takes lots of time
 
   outputs = [
     "bin"
     "dev"
-    "out"
-    "man"
     "doc"
+    "man"
+    "out"
   ];
 
-  meta = with lib; {
+  nativeInstallCheckInputs = [ versionCheckHook ];
+  doInstallCheck = true;
+  versionCheckProgramArg = "--version";
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
     homepage = "https://xiph.org/flac/";
     description = "Library and tools for encoding and decoding the FLAC lossless audio file format";
-    changelog = "https://xiph.org/flac/changelog.html";
+    changelog = "https://github.com/xiph/flac/releases/tag/${finalAttrs.version}";
     mainProgram = "flac";
-    platforms = platforms.all;
-    license = licenses.bsd3;
-    maintainers = with maintainers; [ ruuda ];
+    platforms = lib.platforms.all;
+    license = with lib.licenses; [
+      bsd3
+      fdl13Only
+      gpl2Only
+      lgpl21Only
```

Comment:
In [at least this file](https://github.com/xiph/flac/blob/1507800de4b70e21be71f38caa0d9079d0bc6e45/include/share/grabbag.h#L5-L8), it’s LGPL 2.1 or later.